clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -7745,6 +7745,16 @@ void Sema::AddSYCLIntelNumBanksAttr(Decl *D, const AttributeCommonInfo &CI,
       }
     }
 
+    if (auto *VD = dyn_cast<VarDecl>(D)) {
+      if (!(VD->getKind() != Decl::ImplicitParam &&
+            VD->getKind() != Decl::NonTypeTemplateParm &&
+           ((VD->getStorageClass() == SC_Static || VD->hasLocalStorage()) ||
+            (isTypeDecoratedWithDeclAttribute<SYCLDeviceGlobalAttr>(VD->getType())
+             || VD->getType().isConstQualified())))){
```

Comment:
I think this is missing the opencl_constant AS check.

Reply:
Thanks for catching! Done